function afficherMonHumeur() {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      if (Math.random() <= 0.6) {
        resolve("Content que √ßa soit bient√¥t le weekend. üòÑ");
      } else {
        reject("Trop mal la t√™te. üò≠");
      }
    }, 2000);
  });
}

afficherMonHumeur()
  .then((message) => console.log("message :>> ", message))
  .catch((error) => console.log("error :>> ", error))
  .finally(() => console.log("L'affiche de l'humeur est termin√©."));

// 2e fa√ßon avec async / await
window.addEventListener("DOMContentLoaded", async () => {
  try {
    const message = await afficherMonHumeur();
    console.log("message :>> ", message);
  } catch (error) {
    console.log("error :>> ", error);
  }

  console.log("L'affiche de l'humeur est termin√©.");
});

// Echainement de promesses

function faireLeCafe() {
  return new Promise((resolve, reject) => {
    if (Math.random() <= 0.8) {
      resolve("Le caf√© coule.");
    } else {
      reject("Plus de filtre.");
    }
  });
}

function seLaver() {
  return new Promise((resolve, reject) => {
    if (Math.random() <= 0.6) {
      resolve("Chante sous la douche");
    } else {
      reject("Plus de shampoing");
    }
  });
}

function dejeuner() {
  return new Promise((resolve, reject) => {
    if (Math.random() <= 0.3) {
      resolve("Miam les c√©r√©ales");
    } else {
      reject("Snif. Plus de c√©r√©ales.");
    }
  });
}


faireLeCafe()
  .then(message => {
    console.log('message :>> ', message);
    return seLaver();
  })
  .then(dejeuner)
  .catch(err => console.log('err :>> ', err))
  .finally(() => console.log("La journ√©e peut commencer."));